### PR TITLE
[ipam-docs] Restructured Documentation: New Unified Documentation Website

### PR DESCRIPTION
**Pull request by pandafy**
Related to https://github.com/openwisp/openwisp-docs/issues/107.
These documents are pulled by the logic being worked on in https://github.com/openwisp/openwisp-docs/pull/189.
We are doing similar changes in all modules.

### CODE REVIEW

File: docs/developer/extending.rst [hunk 1]
```diff
@@ -0,0 +1,328 @@
+Extending openwisp-ipam
```
Comment (nemesifier) [inline docs/developer/extending.rst:1]:
```suggestion
Extending OpenWISP IPAM
```

File: README.rst [hunk 2]
```diff
@@ -1,817 +1,54 @@
...
+    :alt: PyPI
 
 .. image:: https://img.shields.io/librariesio/release/github/openwisp/openwisp-ipam
-  :target: https://libraries.io/github/openwisp/openwisp-ipam#dependencies
-  :alt: Dependency monitoring
+    :target: https://libraries.io/github/openwisp/openwisp-ipam#dependencies
+    :alt: Dependency monitoring
 
 .. image:: https://github.com/openwisp/openwisp-ipam/raw/docs/docs/subnet_demo.gif
-  :alt: Feature Highlights
```
Comment (nemesifier) [inline README.rst:22]: why remove the alt?

File: README.rst [hunk 3]
```diff
@@ -1,817 +1,54 @@
...
-==================
-
-View to export subnet data.
+    :alt: Feature Highlights
 
-POST
-----
+Refer to `IPAM: Features
+<https://openwisp.io/docs/__new__/dev/ipam/user/intro.html>` of OpenWISP
+documentation for a complete overview of features.
```
Comment (nemesifier) [inline README.rst:25]: This sentence doesn't sound right to me, have you run it through AI tools to make it better?

Remember: the structure of the READMEs need to be similar, I think we are missing the links to the usage docs and developer docs here.

File: docs/index.rst [hunk 4]
```diff
@@ -0,0 +1,26 @@
+IPAM
+====
+
+.. image:: https://github.com/openwisp/openwisp-ipam/raw/docs/docs/subnet_demo.gif
+    :target: https://github.com/openwisp/openwisp-ipam/raw/docs/docs/subnet_demo.gif
+    :alt: Feature Highlights
+
+OpenWISP IPAM is designed to offer essential IP Address Management (IPAM)
+functionalities as a reusable Django application. It seamlessly integrates
+with the broader OpenWISP ecosystem.
```
Comment (nemesifier) [inline docs/index.rst:10]: This part does not convince me, what do you think of the following alternative?

```
OpenWISP IPAM provides IP Address Management (IPAM) features, refer to :doc:`user/intro` for a complete overview.
As a core dependency of the OpenWISP Controller, it facilitates the automatic provisioning of IP addresses for VPNs 
such as Wireguard and Zerotier, and allows to implement the Subnet Division Rules feature (EDITOR NOTE: please link those pages).

In addition to its integration with the OpenWISP ecosystem, OpenWISP IPAM can be used as a standalone Django app. 
Developers proficient in Python and Django can leverage this module independently to enhance their projects.
For more details, please refer to the developer documentation (LINK).
```

**Review by nemesifier — CHANGES_REQUESTED** [review]
Getting this warning:

```
developer/index.rst:15: WARNING: unknown document: '../user/settings'
```

File: README.rst (hunk 2 above)
Comment (pandafy) [reply README.rst:22]: It is present there, just GitHub showing changes in a weird way.

File: docs/developer/extending.rst [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+- `sample_ipam/__init__.py
+  <https://github.com/openwisp/openwisp-ipam/tree/master/tests/openwisp2/sample_ipam/__init__.py>`_.
+- `sample_ipam/apps.py
+  <https://github.com/openwisp/openwisp-ipam/tree/master/tests/openwisp2/sample_ipam/apps.py>`_.
+
+You have to replicate and adapt that code in your project.
+
+For more information regarding the concept of ``AppConfig`` please refer
+to the `"Applications" section in the django documentation
+<https://docs.djangoproject.com/en/dev/ref/applications/>`_.
```
Comment (nemesifier) [inline docs/developer/extending.rst:156]: please change all the links to point to django 4.2, every time we'll update the recommended django version we'll find & replace across all repos

**Review by nemesifier — CHANGES_REQUESTED** [review]

**Review by nemesifier — APPROVED** [review]
Ready!

**Review by nemesifier — APPROVED** [review]

**Pull request merged**